provisioning/transport/amqp/test/_amqp_test.js

```diff
@@ -446,5 +475,155 @@ describe('Amqp', function () {
+        assert.deepEqual(SaslTpm.firstCall.args[1], new Builder()
+          .appendUInt8(0)
+          .appendString('fakeScope')
+          .appendUInt8(0)
+          .appendString('fakeRegistrationId')
+          .appendUInt8(0)
+          .appendBuffer(fakeEndorsementKey)
+          .get()
+        );
+
+        /*Tests_SRS_NODE_PROVISIONING_AMQP_18_014: [ `getAuthenticationChallenge` shall send the initial response to the AMQP service in the form  "<0><storageRootKey>" where <0> is a zero byte. ]*/
+        assert.deepEqual(SaslTpm.firstCall.args[2], new Builder()
+          .appendUInt8(0)
+          .appendBuffer(fakeStorageRootKey)
+          .get()
+        );
+
+        callback();
+      });
+
+      it ('returns success correctly', function(callback) {
+        amqp.getAuthenticationChallenge(fakeRequest, function(err, challenge) {
+          /*Tests_SRS_NODE_PROVISIONING_AMQP_18_015: [ `getAuthenticationChallenge` shall call `callback` passing `null` and the challenge buffer after the challenge has been received from the service. ]*/
+          assert.isNull(err);
+          assert.strictEqual(challenge.authenticationKey, fakeAuthenticationKey);
+          callback();
+        });
+
+        var challengeCallback = SaslTpm.firstCall.args[3];
+        challengeCallback(fakeAuthenticationKey);
+      });
+
+    });
+
+    describe('respondToAuthenticationChallenge', function() {
+
+      /*Tests_SRS_NODE_PROVISIONING_AMQP_18_017: [ `respondToAuthenticationChallenge` shall call `callback` with an `InvalidOperationError` if called before calling `getAthenticationChallenge`. ]*/
+      it ('fails if called too early', function(callback) {
+        amqp.respondToAuthenticationChallenge(fakeRequest, fakeSasToken, function(err) {
+          assert.instanceOf(err, errors.InvalidOperationError);
+          callback();
+        });
+      });
+
+      /*Tests_SRS_NODE_PROVISIONING_AMQP_18_018: [ `respondToAuthenticationChallenge` shall respond to the auth challenge to the service in the form "<0><sasToken>" where <0> is a zero byte. ]*/
+      it ('responds to the challenge correctly', function(callback) {
+        amqp.getAuthenticationChallenge(fakeRequest, function(err, challenge) {
+          assert.isNull(err);
+          amqp.respondToAuthenticationChallenge(fakeRequest, fakeSasToken, function() {});
+        });
+
+        var challengeCallback = SaslTpm.firstCall.args[3];
+        // return the auth key to the transport, verify that it responds with the sas token above.
+        challengeCallback(fakeAuthenticationKey, function(err, challengeResponse) {
+          assert.deepEqual(challengeResponse, new Builder()
+            .appendUInt8(0)
+            .appendString(fakeSasToken)
+            .get()
+          );
+          callback();
+        });
+      });
+
+      var respondToAuthChallengeCorrectly = function() {
+        var challengeCallback = SaslTpm.firstCall.args[3];
```

Comment:
mocking/stubbing in here is really convoluted. not opposed to testing this way but i wonder if this is code smell.